src/backend/distributed/commands/table.c

```diff
@@ -576,6 +576,19 @@ PreprocessAttachCitusPartitionToCitusTable(Oid parentCitusRelationId, Oid
 		ereport(ERROR, (errmsg("non-distributed partitioned tables cannot have "
 							   "distributed partitions")));
 	}
+	else if (IsCitusTableType(partitionRelationId, CITUS_LOCAL_TABLE) &&
+			 IsCitusTableType(parentCitusRelationId, NULL_KEY_DISTRIBUTED_TABLE))
+	{
+		/*
+		 * The order here is important, because NULL_KEY_DISTRIBUTED_TABLE's are
+		 * also DISTRIBUTED_TABLE's. This 'else if' should come before the next one.
```

Comment:
To get rid of this ambiguity, I think we should incorporate this into below block, as in:
```c
/* if the parent is a distributed table, distribute the partition too */
if (HasDistributionKey(parentCitusRelationId))
  CreateNullShardKeyDistTable(partitionRelationId, parentName);
else:
  DistributePartitionUsingParent(parentCitusRelationId, partitionRelationId);

```

Reply:
Or maybe we should distinguish null-shard-key vs regular tables in DistributePartitionUsingParent?

I think this would be a good practice because if we did so in https://github.com/citusdata/citus/pull/6778, then we wouldn't have this bug at all.